src/main/java/software/amazon/awssdk/crt/http/HttpProxyEnvironmentVariableSetting.java

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+package software.amazon.awssdk.crt.http;
+
+import software.amazon.awssdk.crt.CrtRuntimeException;
+import software.amazon.awssdk.crt.io.TlsContext;
+import software.amazon.awssdk.crt.http.HttpProxyOptions.HttpProxyConnectionType;
+
+/**
+ * This class provides access to Http proxy environment variable configuration
+ * setting
+ */
+public class HttpProxyEnvironmentVariableSetting {
+
+    private TlsContext tlsContext;
+    private HttpProxyEnvironmentVariableType environmentVariableType;
+    private HttpProxyConnectionType connectionType;
+
+    /*
+     * Configuration for using proxy from environment variable.
+     */
+    public enum HttpProxyEnvironmentVariableType {
+
+        /**
+         * Disable reading from environment variable for proxy.
+         */
+        DISABLED(0),
+
+        /**
+         * Default.
+         * Enable reading from environment variable for proxy configuration, when the proxy options
+         * are not set.
+         * env HTTPS_PROXY/https_proxy will be checked when the main connection use tls.
+         * env HTTP_PROXY/http_proxy will be checked when the main connection does not use tls.
+         * The lower case version has precedence.
+         */
+        ENABLED(1);
+
+        private int environmentVariableType;
+
+        HttpProxyEnvironmentVariableType(int val) {
+            environmentVariableType = val;
+        }
+
+        public int getValue() {
+            return environmentVariableType;
+        }
+    }
```

Comment:
This enum name matches the name in the C code. However, I think it is confusing and am trying to think of a better name.